extensions-core/multi-stage-query/src/main/java/org/apache/druid/msq/querykit/WindowOperatorQueryKit.java

```diff
@@ -105,8 +108,20 @@ public QueryDefinition makeQueryDefinition(
 
     final int firstStageNumber = Math.max(minStageNumber, queryDefBuilder.getNextStageNumber());
     final WindowOperatorQuery queryToRun = (WindowOperatorQuery) originalQuery.withDataSource(dataSourcePlan.getNewDataSource());
-    final int maxRowsMaterialized;
 
+    // Get segment granularity from query context, and create ShuffleSpec and RowSignature to be used for the final window stage.
+    final Granularity segmentGranularity = QueryKitUtils.getSegmentGranularityFromContext(jsonMapper, queryToRun.getContext());
+    final ClusterBy finalWindowClusterBy = QueryKitUtils.clusterByWithSegmentGranularity(ClusterBy.none(), segmentGranularity);
```

Comment:
Why is this `ClusterBy.none()`? I am not certain if that's wrong, but we should mention a reason as to why the final stage doesn't sort by anything apart from time.

Reply:
I'm not sure what ClusterBy should be used here. I don't see any ClusterBy to extract from the available fields in this class. Thoughts? Please let me know if I'm missing something. Thanks!

Reply:
I've pushed some logic changes as per offline discussion. Please have a look, will close this comment as it's outdated now. Thanks!